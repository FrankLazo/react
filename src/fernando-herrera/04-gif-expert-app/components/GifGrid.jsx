import PropTypes from 'prop-types';
import GifGridItem from './GifGridItem';
import useFetchGifs from '../hooks/useFetchGifs';
import 'animate.css';

const GifGrid = ({ category }) =>
{
	const { loading, data: images } = useFetchGifs({ category });

	return (
		<div className="has-text-light">
			<h2 className="has-text-light animate__animated animate__fadeIn">{ category }</h2>

		{
			loading && <p className="animate__animated animate__fadeIn">Loading...</p>
		}

			<div className="card-grid">
			{
				images.map(image => (
					<GifGridItem
						key={ image.id }
						{ ...image }
						/>
				))
			}
			</div>
		</div>
	);
};

GifGrid.propTypes = {
	category: PropTypes.string.isRequired,
}

export default GifGrid;
