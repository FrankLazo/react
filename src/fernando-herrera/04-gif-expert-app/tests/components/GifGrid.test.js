import { shallow } from 'enzyme';
import GifGrid from "../../components/GifGrid";
import useFetchGifs from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Tests on GifGrid', () => {
	const category = 'Megadeth';

	test('should show the right component and match with the snapshot', () => {
		useFetchGifs.mockReturnValue({
			data:    [],
			loading: true,
		});

		const wrapper = shallow(<GifGrid category={ category } />);
		expect(wrapper).toMatchSnapshot();
	});

	test('should show items when load images with useFetchGifs', () => {
		const gifs = [{
			id:    'abc',
			url:   'https://localhost/anything.jpg',
			title: 'Anything',
		},];

		useFetchGifs.mockReturnValue({
			data:    gifs,
			loading: false,
		});

		const wrapper = shallow(<GifGrid category={ category } />);

		expect(wrapper).toMatchSnapshot();
		expect( wrapper.find('p').exists() ).toBe(false);
		expect( wrapper.find('GifGridItem').length).toBe(gifs.length);
	});

});
