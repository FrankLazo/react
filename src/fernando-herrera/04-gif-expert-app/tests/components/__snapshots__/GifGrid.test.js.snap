// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on GifGrid should show items when load images with useFetchGifs 1`] = `
<div
  className="has-text-light"
>
  <h2
    className="has-text-light animate__animated animate__fadeIn"
  >
    Megadeth
  </h2>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Anything"
      url="https://localhost/anything.jpg"
    />
  </div>
</div>
`;

exports[`Tests on GifGrid should show the right component and match with the snapshot 1`] = `
<div
  className="has-text-light"
>
  <h2
    className="has-text-light animate__animated animate__fadeIn"
  >
    Megadeth
  </h2>
  <p
    className="animate__animated animate__fadeIn"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</div>
`;
